File: 2017m./uzd2./uzd2.cpp
#include <iostream>
#include <fstream>
#include <vector>

using namespace std;

class Pixel {
public:
    int r;
    int g;
    int b;

    Pixel(int r = 255, int g = 255, int b = 255) : r(r), g(g), b(b) {}
};

vector<vector<Pixel>> placeSquare(int x, int y, int offsetX, int offsetY, vector<vector<Pixel>>& image, const Pixel& newPixel) {
    int endPosX = x + offsetX;
    int endPosY = y + offsetY;

    if (endPosX - 1 >= image[0].size()) {
        int pixelsToAdd = endPosX - image[0].size();
        for (auto& row : image) {
            row.insert(row.end(), pixelsToAdd, Pixel());
        }
    }

    if (endPosY - 1 >= image.size()) {
        image.resize(endPosY, vector<Pixel>(image[0].size(), Pixel()));
    }

    for (int i = y; i < endPosY; ++i) {
        for (int j = x; j < endPosX; ++j) {
            image[i][j] = newPixel;
        }
    }

    return image;
}

int main() {
    vector<vector<Pixel>> image(1, vector<Pixel>(1, Pixel()));

    ifstream data("U2.txt");
    ofstream output("U2rez.txt");

    int squareCount;
    data >> squareCount;

    for (int i = 0; i < squareCount; ++i) {
        int x, y, offsetX, offsetY, r, g, b;
        data >> x >> y >> offsetX >> offsetY >> r >> g >> b;
        Pixel newPixel(r, g, b);
        image = placeSquare(x, y, offsetX, offsetY, image, newPixel);
    }

    output << image.size() << " " << image[0].size() << "\n";

    for (const auto& row : image) {
        for (const Pixel& pixel : row) {
            output << pixel.r << " " << pixel.g << " " << pixel.b << " " << "\n";
        }
    }

    data.close();
    output.close();

    return 0;
}

File: 2017m./uzd2./uzd2.py
class Pixel:
    def __init__(self, r = 255, g = 255, b = 255):
        self.r = r
        self.g = g
        self.b = b

def placeSquare(x, y, offsetX, offsetY, image, newPixel):
    endPosX = x + offsetX
    endPosY = y + offsetY
    
    if(endPosX - 1 >= len(image[0])):
        pixelsToAdd = endPosX - len(image[0])
        for i in range(len(image)):
            image[i] = image[i] + [Pixel() for i in range(pixelsToAdd)]
    
    if(endPosY - 1 >= len(image)):
        image.append([Pixel() for j in range(len(image[0]))])

    for i in range(y, endPosY):
        for j in range(x, endPosX):
            image[i][j] = newPixel
    
    return image

def main():
    image = [[Pixel()]]

    data = open("U2.txt", "r")
    output = open("U2rez.txt", "w")

    squareCount = int(next(data))

    for i in range(squareCount):
        newSquare = next(data).split(" ")
        newSquare = [int(i) for i in newSquare]
        newPixel = Pixel(newSquare[4], newSquare[5], newSquare[6])
        image = placeSquare(newSquare[0], newSquare[1], newSquare[2], newSquare[3], image, newPixel)

    output.write(str(len(image)) + " " + str(len(image[0])) + "\n")

    for i in range(len(image)):
        for j in range(len(image[0])):
            output.write(str(image[i][j].r) + " " + str(image[i][j].g) + " " + str(image[i][j].b) + " " "\n")
    
    data.close()
    output.close()

if __name__ == "__main__":
    main()